client/components/BotRunTestStatusList/index.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    COLLECTION_JOB_STATUS_BY_TEST_PLAN_RUN_ID_QUERY,
+    TEST_PLAN_RUNS_TEST_RESULTS_QUERY
+} from './queries';
+import { useLazyQuery, useQuery } from '@apollo/client';
+import styled from '@emotion/styled';
+import ReportStatusDot from '../common/ReportStatusDot';
+import { isBot } from '../../utils/automation';
+
+const BotRunTestStatusUnorderedList = styled.ul`
+    list-style-type: none;
+    background-color: #f6f8fa;
+    font-size: 0.9rem !important;
+    padding: 0.5rem 0;
+    margin: 0.5rem 0;
+    white-space: nowrap;
+`;
+
+const BotRunTestStatusList = ({ testPlanReportId, runnableTestsLength }) => {
+    const {
+        data: testPlanRunsQueryResult,
+        startPolling,
+        stopPolling
+    } = useQuery(TEST_PLAN_RUNS_TEST_RESULTS_QUERY, {
+        variables: { testPlanReportId },
+        fetchPolicy: 'cache-and-network',
+        pollInterval: 2000
+    });
+
+    const [getCollectionJobStatus, { data: collectionJobStatusQueryResult }] =
+        useLazyQuery(COLLECTION_JOB_STATUS_BY_TEST_PLAN_RUN_ID_QUERY, {
```

Comment:
Beyond the context of this patch, I'd like to pick your brain about this pattern.

Reply:
Sounds good! This was an initial pass on "real-time" updates before I found out this would be a focus in v2. I do think the pattern should be improved and possibly refactored. I didn't spend additional cycles on it due to the aforementioned v2 focus.